aea/configurations/base.py

```diff
@@ -2099,17 +2101,30 @@ def json(self) -> Dict:
                 },
             }
         )
+        if self.keep_terminal_state_dialogues is not None:
+            result["keep_terminal_state_dialogues"] = self.keep_terminal_state_dialogues
```

Comment:
see above; prefer to have a default as we role this in as a new feature now; not an optional one

Reply:
done